docs/concepts/storage.md

```diff
@@ -3,3 +3,157 @@ sidebar_position: 3
+
+:::
+
+## History & Background
+
+Historically, Kafka Streams bundled storage locally on compute nodes by
+providing RocksDB based implementations of state stores. While storage on the
+compute nodes was not required to be durable (it could be recovered from
+a [changelog topic](https://developer.confluent.io/courses/kafka-streams/stateful-fault-tolerance/)
+in Kafka), performance necessitated storage to be persistent across restarts.
+The motivation for such a design was clear at the time: most deployments of
+Kafka Streams were running without access to the public cloud, and keeping
+the library free of external dependencies other than Kafka made it uniquely
+easy to deploy.
+
+The limitations of copuling compute with storage are well known: 
+1. Sizing your nodes becomes more of an art than a science, and many Kafka
+   Streams deployments in the wild tend to overprovision on both to ensure
+   that neither limitation is met.
+1. Scaling out becomes difficult since new nodes need to bootstrap storage
+   before they can become operational.
+1. Recovering from a fault or rebalancing is a challenge for the same reason
+   as above.
+
+:::info
+
+Curious to read more about tuning RocksDB state stores and contrasting it with
+Responsive Storage? Read our [blog post](https://responsive.dev/blog/separating-storage-and-compute)
+about it.
+
+:::
+
+Over time, the Kafka Streams developers have come up with elegant improvements
+to these problems: standby and warmup replicas, incremental rebalancing, sticky 
+task assignment... the list goes on.  Responsive takes a different approach to 
+solving this class of problems in Kafka Streams by improving the underlying 
+architecture to separate compute and storage - effectively eliminating these 
+concerns in one fell swoop.
+
+## Remote Storage
+
+
+At Responsive, we believe that reactive backends should power the most business-
+critical applications in your stack, that's why we built our storage solution
+to leverage [Scylla DB](https://github.com/scylladb/scylladb), a battle-tested
+distributed database that provides:
+
+- Fast writes & fast Key-Value reads
+- The transactional guarantees to implement the EOS `processing.guarantee`
+- Support for range queries
+- World class operations: scalability, durability, fault-tolerance
+
+But just as Kafka Streams must solve many tough problems despite delegating
+the storage layer to RocksDB, Responsive has developed technology to ensure
+that Scylla operates well in the context of an event processing framework. The
+next few sections examine some of the concepts we've implemented.
+
+### Instant Restoration
+
+With a RocksDB-based deployment of Kafka Streams, restarting a node that has
+both a RocksDB store and a checkpoint file is easy: the application will know
+where from the changelog to start reading from and skip to that point before
+starting restoration.
+
+Responsive's remote storage takes inspiration from that strategy, and makes
```

Comment:
I think we can go into more detail on how we use the changelog to leverage kafka transactions and exactly once, while not requiring long restore times.

I also think we should talk about how we have managed decoupling the partitioning for storage from the partitioning of the kafka topics, which should allow easier repartitioning of Kafka topics.

Reply:
I'll add a section on the latter - that's a great call, the former is pretty deep and I think if we wanted we could just write an entire section or a blog about it - I don't think it makes sense to go into detail of our EOS design here.